contracts/contracts/coordination/Subscription.sol

```diff
@@ -0,0 +1,261 @@
+    mapping(bytes32 => uint32) public subscribers;
+
+    uint32 public numberOfSubscriptions;
+
+    // TODO: DAO Treasury
+    address public beneficiary;
+
+    /**
+     * @notice Emitted when a subscription is created
+     * @param subscriptionId The ID of the subscription
+     * @param subscriber The address of the subscriber
+     * @param ritualId The ID of the ritual
+     */
+    event SubscriptionCreated(
+        uint32 indexed subscriptionId,
+        address indexed subscriber,
+        uint32 indexed ritualId
+    );
+
+    /**
+     * @notice Emitted when a subscription is cancelled
+     * @param subscriptionId The ID of the subscription
+     * @param subscriber The address of the subscriber
+     * @param ritualId The ID of the ritual
+     */
+    event SubscriptionCancelled(
+        uint32 indexed subscriptionId,
+        address indexed subscriber,
+        uint32 indexed ritualId
+    );
+
+    /**
+     * @notice Emitted when a subscription is paid
+     * @param subscriptionId The ID of the subscription
+     * @param subscriber The address of the subscriber
+     * @param amount The amount paid
+     */
+    event SubscriptionPaid(
+        uint32 indexed subscriptionId,
+        address indexed subscriber,
+        uint256 amount
+    );
+
+    /**
+     * @notice Sets the coordinator and fee token contracts
+     * @dev The coordinator and fee token contracts cannot be zero addresses
+     * @param _coordinator The address of the coordinator contract
+     * @param _feeToken The address of the fee token contract
+     * @param _beneficiary The address of the beneficiary
+     */
+    constructor(Coordinator _coordinator, IERC20 _feeToken, address _beneficiary) {
+        require(address(_coordinator) != address(0), "Coordinator cannot be the zero address");
+        require(address(_feeToken) != address(0), "Fee token cannot be the zero address");
+        require(_beneficiary != address(0), "Beneficiary cannot be the zero address");
+        coordinator = _coordinator;
+        feeToken = _feeToken;
+        beneficiary = _beneficiary;
+    }
+
+    /**
+     * @notice Returns the subscription fee
+     * @return The subscription fee
+     */
+    function subscriptionFee() public pure returns (uint256) {
+        return 42 * 10 ** 20; // TODO
+    }
+
+    /**
+     * @notice Returns the base expiration duration
+     * @return The base expiration duration
+     */
+    function baseExpiration() public pure returns (uint256) {
+        return 52 weeks; // TODO
+    }
+
+    /**
+     * @notice Returns the key used to lookup authorizations
+     * @param ritualId The ID of the ritual
+     * @param encryptor The address of the encryptor
+     * @return The key used to lookup authorizations
+     */
+    function lookupKey(uint32 ritualId, address encryptor) internal pure returns (bytes32) {
+        return keccak256(abi.encodePacked(ritualId, encryptor));
+    }
```

Comment:
I suggest to extract this method to library or to just some common contract unless it's too completely different key building here and in allow list